src/charm.py

```diff
@@ -122,12 +125,16 @@ def _on_start(self, event: StartEvent) -> None:
             self._mysql.configure_mysql_users()
             self._mysql.configure_instance()
             self._mysql.wait_until_mysql_connection()
+            workload_version = self._mysql.get_mysql_version()
+            self.unit.set_workload_version(workload_version)
         except MySQLConfigureMySQLUsersError:
             self.unit.status = BlockedStatus("Failed to initialize MySQL users")
             return
         except MySQLConfigureInstanceError:
             self.unit.status = BlockedStatus("Failed to configure instance for InnoDB")
             return
+        except MySQLClientError:
```

Comment:
1. our pattern has been to not catch `MySQLClientError` in the charm code
2. shouldnt the charm go into blockedstatus?

Reply:
1. Ok
2. not really. this is an optional field